crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1159,9 +1181,50 @@ impl fmt::Debug for World {
     }
 }
 
-unsafe impl Send for World {}
 unsafe impl Sync for World {}
 
+/// A world which does not contain any [`!Send`] resources, and therefore
+/// can be safely sent between threads.
+///
+/// The name turtle is derived from this being something which is moving a
+/// [`World`] (between threads)
+///
+/// [`!Send`]: Send
+#[derive(Debug)]
+pub struct Turtle {
+    // Safety: does not have any !Send resources
+    world: World,
+}
+
+// Safety: The contained world does not contain anything which is !Send
+unsafe impl Send for Turtle {}
+
+impl Turtle {
+    /// The [`World`] this [`Turtle`] was created from.
+    ///
+    /// The returned [`World`] does not contain any [`!Send`] resources
+    ///
+    /// [`!Send`]: Send
+    pub fn world(self) -> World {
+        self.world
+    }
+
+    /// A view on this [`Turtle`]'s [`World`], which contains no [`!Send`] resources
+    ///
+    /// [`!Send`]: Send
+    // Safety: NonSend resources cannot be added using a raw reference
```

Comment:
this is a \_really\_ sketchy safety comment imo, its relying on things not being possible in the current API which could definitely be added in the future. Do we actually need this function? It doesn't seem like it from the changes in this PR

Reply:
How would having a raw reference to the world ever allow adding `!Send` resources.

Also how would you propose we access the world in a scene without this method?

Reply:
Well resources have no need to fragment archetypes, which is the main problem with `&World` allowed add/removes on components.

After looking over the code again I realise that you have `&Turtle` not `Turtle` so yeah I dunno what to do here but I think this safety comment is not going to end well

Reply:
Also, "raw" reference feels like poor wording, maybe immutable reference or just reference would work better?

Reply:
Yeah, not sure why I used the term raw here.

Also, in my view, the main challenge of `&World` allowing adds and removes is actually synchronisation; currently the synchronisation is outside of the `World` (for good reason). You seem to be suggesting that there are plans to add the synchronisation back into the `World`?